### PR TITLE
feat(cache): add project column

### PR DESCRIPTION
**Pull request by DominikB2014**
Adds project column to cache module and updates http module to use the existing field renderer now that we have `project` field in discover for spanMetrics https://github.com/getsentry/sentry/pull/69703

image

image

One thing to note, the icon for the project is smaller then what we currently have, however, this matches our other tables in discover and performance, so it will be more consistent.

### CODE REVIEW

File: static/app/views/performance/cache/tables/transactionsTable.tsx [hunk 1]
```diff
@@ -34,7 +34,12 @@ type Row = Pick<
 >;
 
 type Column = GridColumnHeader<
-  'transaction' | 'spm()' | 'cache_miss_rate()' | 'time_spent_percentage()'
+  | 'transaction'
+  | 'spm()'
+  | 'cache_miss_rate()'
+  | 'time_spent_percentage()'
+  | 'project.id'
```
Comment (gggritso) [inline static/app/views/performance/cache/tables/transactionsTable.tsx:41]:
```suggestion
  | 'project.id'
```

Not needed anymore, right?

File: static/app/views/starfish/types.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export enum SpanMetricsField {
   SPAN_GROUP = 'span.group',
   SPAN_DURATION = 'span.duration',
   SPAN_SELF_TIME = 'span.self_time',
+  PROJECT = 'project',
```
Comment (gggritso) [inline static/app/views/starfish/types.tsx:30]: Could you add a note here that this is the ID, and not the slug? Same in `SpanIndexedField` below

File: static/app/views/starfish/types.tsx
Comment (gggritso) [inline static/app/views/starfish/types.tsx:1]: Could you update the `MetricsResponse` type below to include `project`? Right next to where it defines `project.id` would be fine 👍🏻

**Review by gggritso — COMMENTED** [review]
B-e-a utiful!

File: static/app/views/starfish/types.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/starfish/types.tsx:30]: My bad! In that case, leave a note that it's the slug!

File: static/app/views/starfish/types.tsx
Comment (DominikB2014) [reply static/app/views/starfish/types.tsx:1]: done!

File: static/app/views/performance/cache/tables/transactionsTable.tsx (hunk 1 above)
Comment (DominikB2014) [reply static/app/views/performance/cache/tables/transactionsTable.tsx]: 👍

File: static/app/views/performance/http/httpLandingPage.tsx [hunk 3]
```diff
@@ -108,7 +108,7 @@ export function HTTPLandingPage() {
   const domainsListResponse = useSpanMetrics({
     search: MutableSearch.fromQueryObject(tableFilters),
     fields: [
-      'project.id',
+      'project',
```
Comment (gggritso) [inline static/app/views/performance/http/httpLandingPage.tsx:111]: 🤔 I think you need both here, or the links will break, no? The `project.id` is used to construct the URL in the link

Comment (DominikB2014) [reply static/app/views/performance/http/httpLandingPage.tsx:111]: Link works for me, looks like the `<Projects>` component internally figures out the id's from the slug

https://github.com/getsentry/sentry/blob/5bdbbb6a3b758cfd7ddd139444a822b9ceceefc1/static/app/utils/discover/fieldRenderers.tsx#L308

Comment (gggritso) [reply static/app/views/performance/http/httpLandingPage.tsx:111]: I mean the link to the domain summary page, rather than the link to the project, sorry!

Comment (DominikB2014) [reply static/app/views/performance/http/httpLandingPage.tsx:111]: Yes you're right! Nice catch! I'll add that back for this PR.

Might be an unnecessary optimization, but I do wonder if we should/could also just pull id from the slug in the DomainCell component. Looking at the projects component, it appears that all the projects are stored in some local state.

Comment (DominikB2014) [reply static/app/views/performance/http/httpLandingPage.tsx]: Added `project.id` back for http and cache to the query (Cache also needs the project.id for a similar reason)

Comment (gggritso) [reply static/app/views/performance/http/httpLandingPage.tsx]: Definitely maybe! There are a lot of shenanigans with pulling project slugs and IDs from the backend

**Review by gggritso — APPROVED** [review]

**Pull request merged**